src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/HostAlignmentReadFilter.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.List;
+
+/**
+ * Filters out reads above a threshold coverage (number of matches, mismatches, and insertions) and identity
+ *   (number of matches less deletions), each given in bases.
+ */
+public final class HostAlignmentReadFilter extends ReadFilter {
+
+    private static final long serialVersionUID = 1L;
+
+    private final int minCoverage, minIdentity;
+
+    public HostAlignmentReadFilter(final int minCov, final int minIdent) {
+        this.minCoverage = minCov;
+        this.minIdentity = minIdent;
+    }
+
+    @Override
+    public boolean test(final GATKRead read) {
+        return read.isUnmapped() || !read.hasAttribute("NM")
```

Comment:
I got the logic of this one on first reading!  So that's good.  But if you break it up, you can avoid the double attribute lookup.
```
if ( read.isUnmapped() ) return true;
final Integer nmVal = read.getAttributeAsInteger("NM");
if ( nmVal == null ) return true;
return test(read.getCigar(), nmVal.intValue());
```
Up to you.  This is somewhat a matter of taste.